### PR TITLE
Fix corner cases for delimiters with more than one character

### LINKED ISSUE #37: Double-check delimiters with nchar > 1
It would be simpler to only accept delimiters of length 1.

### PR DESCRIPTION
**Pull request by krlmlr**
Currently contains only failing tests.

Fixes #37.

### CODE REVIEW

File: tests/testthat/test-glue.R [hunk 1]
```diff
@@ -267,8 +267,15 @@ glue("
 
 test_that("glue works with alternative delimiters", {
   expect_identical(as_glue("{1}"), glue("{1}", .open = "", .close = ""))
+  expect_identical(as_glue("{{}}"), glue("{{}}", .open = "", .close = ""))
 
   expect_identical(as_glue("1"), glue("<<1>>", .open = "<<", .close = ">>"))
+  expect_identical(as_glue("<<<<>>>>"), glue("<<<<>>>>", .open = "<<", .close = ">>"))
```
Comment (jimhester) [inline tests/testthat/test-glue.R]: This test is wrong, doubled delimiters are converted to singular in the output, so it should be testing against `as_glue("<<>>")`

**Comment by jimhester** [comment]
I merged these tests manually and fixed the failing ones, so I am closing this.

**Comment by krlmlr** [comment]
Thanks. I'm not sure I understand the code in the first state of the FSM, and its behavior:

``` r
library(glue)
glue("}}", .close = "}}")
#> }}
glue("}}}", .close = "}}")
#> }}}
glue("}}}}", .close = "}}")
#> }}
glue("}}}}}", .close = "}}")
#> }}}
glue("}}}}}}", .close = "}}")
#> }}}}
glue("}}}}}}}", .close = "}}")
#> }}}
```

**Comment by krlmlr** [comment]
But then these corner cases may be of little relevance.

**Pull request closed without merging**